services-core/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -209,6 +189,48 @@ impl UpdatingOrderbook {
         }
         Ok(())
     }
+
+    async fn chunked_events(
+        &self,
+        from_block: u64,
+        to_block: u64,
+    ) -> Result<impl Stream<Item = Result<Vec<Event>, ExecutionError>> + '_> {
+        let event_stream = self
+            .contract
+            .past_events(
+                BlockNumber::Number(from_block.into()),
+                BlockNumber::Number(to_block.into()),
+                self.block_page_size as _,
+            )
+            .await?;
+        let event_chunks = event_stream
+            .ready_chunks(self.block_page_size)
+            .map(|chunk| chunk.into_iter().collect::<Result<Vec<_>, _>>());
+        Ok(event_chunks)
+    }
+
+    // Retrieve the needed timestamps using a batch transport.
+    async fn prepare_timestamp_cache(
+        &self,
+        context: &mut Context,
+        events: &[Event],
+        latest_block: u64,
+    ) -> Result<()> {
+        let block_hashes = events
+            .iter()
+            .map(|event| {
+                let metadata = event
+                    .meta
+                    .as_ref()
+                    .ok_or_else(|| anyhow!("event without metadata: {:?}", event))?;
+                Ok(metadata.block_hash)
+            })
+            .collect::<Result<HashSet<H256>>>()?;
+        context
+            .block_timestamp_reader
+            .prepare_cache(block_hashes, self.block_page_size, latest_block)
```

Comment:
Do we still need block_page_size in prepare cache? If so, should it be `events.len()` (I see it's the same value but this is not clear from just looking at `prepare_timestamp_cache` in isolation).

Reply:
It's not *needed* but I feel it still makes sense because the batch size we use for events on the updating_orderbook layer doesn't have to be the same batch size as for the CachedBlockTimestampReading layer.
You could argue that with the current code it is clear what the batch size whereas if we use `block_hashes.len()` then you have to look at how the function is called to figure out how many block hashes there are.

Reply:
I also prefer slightly using `block_page_size`, this also makes it a bit more resilient to refactors (when we may call `prepare_timestamp_cache` with more events).